db-engine/src/main/java/com/splicemachine/db/impl/jdbc/EmbedConnection.java

```diff
@@ -112,7 +112,9 @@
      */
     public static final SQLException NO_MEM =
         Util.generateCsSQLException(SQLState.LOGIN_FAILED, "java.lang.OutOfMemoryError");
-    
+
+	public static final ThreadLocal<Boolean> isHBaseJVM = new ThreadLocal<>();
```

Comment:
I don't think this should be a ThreadLocal, are we setting this property for every Thread that could access this field? If  a NLJ runs in a different thread that might not have executed bootDatabase() it will think it's not in a regionServer. This should be either a static volatile boolean / static AtomicBoolean or we'd need a different mechanism.

Reply:
This field is only set and used during updateLoader initialization, and set a flag for the singleton UpdaterLoader whether it is running on a HBase. If any thread needs to load jar, it should use the shared singleton UpdateLoader instance, which always know where it is running. No thread needs to access the ThreadLocal variable and decide whether it is running on HBase

Reply:
I see, thanks @jyuanca !